src/style/style.js

```diff
@@ -375,7 +375,9 @@ class Style extends Evented {
 
         if (!projectionChanged) return;
 
-        if (projection.isReprojectedInTileSpace || prevProjection.isReprojectedInTileSpace) {
+        const globeChanged = projection.name === 'globe' || prevProjection.name === 'globe';
+
+        if (projection.isReprojectedInTileSpace || prevProjection.isReprojectedInTileSpace || globeChanged) {
```

Comment:
I'm curious as to why it's necessary from this PR. Do we need an extra clear for circles?

If we go through `clearBackgroundTiles` we introduce a flicker when implicitly switching the projection from globe to mercator and back:

https://user-images.githubusercontent.com/7061573/151275384-86d280a1-5c06-4957-a68a-c180025f4824.mp4

If it's not necessary we should try to only go through `_forceSymbolLayerUpdate()` to prevent that.

Reply:
Ah didn't notice the flickering. This is a good find!

I do think we need a way to trigger relayout per source in order to reload (and hence reproject) circle layers when projection changes. This should be done only on explicit changes though.

![globe_circle_bug](https://user-images.githubusercontent.com/55925868/151355375-b145d012-a20a-4202-a12a-46be09b2da1f.gif)